packages/nextjs/utils/scaffold-eth/contract.ts

```diff
@@ -231,12 +232,44 @@ export type EventFilters<
 export type UseScaffoldEventHistoryConfig<
   TContractName extends ContractName,
   TEventName extends ExtractAbiEventNames<ContractAbi<TContractName>>,
+  TBlockData extends boolean = false,
+  TTransactionData extends boolean = false,
+  TReceiptData extends boolean = false,
 > = {
   contractName: TContractName;
   eventName: IsContractDeclarationMissing<string, TEventName>;
   fromBlock: bigint;
   filters?: EventFilters<TContractName, TEventName>;
-  blockData?: boolean;
-  transactionData?: boolean;
-  receiptData?: boolean;
+  blockData?: TBlockData;
+  transactionData?: TTransactionData;
+  receiptData?: TReceiptData;
 };
+
+export type UseScaffoldEventHistoryData<
+  TContractName extends ContractName,
+  TEventName extends ExtractAbiEventNames<ContractAbi<TContractName>>,
+  TBlockData extends boolean = false,
+  TTransactionData extends boolean = false,
+  TReceiptData extends boolean = false,
+  TEvent extends ExtractAbiEvent<ContractAbi<TContractName>, TEventName> = ExtractAbiEvent<
+    ContractAbi<TContractName>,
+    TEventName
+  >,
+> = IsContractDeclarationMissing<
+  any[] | undefined,
```

Comment:
You also can use `IsContractDeclarationMissing<any[], {...}> | undefined` instead of `| undefined` in two places. Or you want to do it more explicit?

Reply:
Yes that looks more better updated at [b7c832a](https://github.com/scaffold-eth/scaffold-eth-2/pull/553/commits/b7c832aaa9ea724a3f4d106c54d2335a8b3083ec), Thanks 🙌